### ISSUE TITLE
Is an item's url required or optional?

### ISSUE THREAD

**Issue by eliahhecht** [brentsimmons/JSONFeed#70]
The spec says:

> url (optional, string) is the URL of the resource described by the item. It’s the permalink. This may be the same as the id — but should be present regardless.

Given the phrase "should be present regardless," is the url actually required? Or am I misreading something?

**Comment by manton** [comment]
The only reason `url` isn't technically required is because you might have offline feeds that don't necessarily point to a place on the web. For any normal feed for a blog or other web site, `url` should be there. The "regardless" just means that if `id` is a URL, it's not enough by itself; you also want `url` to have the same value. (In other words, clients shouldn't assume that `id` is a URL or be required to treat it as one.)

**Comment by eliahhecht** [comment]
I see. Maybe rephrase to something like "This may be the same as the id, but even if it is, it should still be present"?

**Comment by dret** [comment]
i am wondering: since JSON feed claims to be JSON-ified atom, what's the corresponding atom element here? it would be great to actually link the atom elements from the spec, as long as there is a strictly corresponding atom element.

**Comment by manton** [comment]
I think the Atom comparison is mostly covered in the "mapping RSS and Atom" document: https://jsonfeed.org/mappingrssandatom
